Hierarchical3DRegistration/Hierarchical3DRegistration.py

```diff
@@ -172,78 +207,196 @@ def initializeParameterNode(self) -> None:
+                                                    f"'{nextBone.name}' in frame {self._parameterNode.currentFrame}"
+                slicer.app.processEvents()
+
+    def onRegisterButton(self):
+        with slicer.util.tryWithErrorDisplay("Failed to compute results.", waitCursor=True):
+            currentRootIDStatus = self.ui.SubjectHierarchyComboBox.currentItem() != 0
+            # TODO: Remove this once this is working with the parameterNodeWrapper.
+            #  It's currently commented out due to bug with parameter node, see
+            #  Slice issue: https://github.com/Slicer/Slicer/issues/7905
+            if not currentRootIDStatus:
+                raise ValueError("Invalid hierarchy object selected!")
+            if self._parameterNode.runSatus == Hierarchical3DRegistrationRunStatus.CANCELING:
+                raise ValueError("Canceling registration...")
+
+            self.currentBone = self.bonesToTrack.pop(0)
+            self._parameterNode.currentBoneID = -1  # TOOD: want this meaningfully saved to scene and re-imported later
+
+            self.currentBone.stopInteraction(self._parameterNode.currentFrame)
 
-                    CT = self.ui.inputSelectorCT.currentNode()
-                    rootID = self.ui.SubjectHierarchyComboBox.currentItem()
+            source_frame = self._parameterNode.currentFrame - 1
+            target_frame = self._parameterNode.currentFrame
 
-                    startFrame = self.ui.startFrame.value
-                    endFrame = self.ui.endFrame.value
+            slicer.util.forceRenderAllViews()
 
-                    trackOnlyRoot = self.ui.onlyTrackRootNodeCheckBox.isChecked()
+            if source_frame < self._parameterNode.startFrame:
+                source_frame = "source"
 
-                    self.logic.registerSequence(CT, rootID, startFrame, endFrame, trackOnlyRoot)
-                finally:
-                    self.inProgress = False
-            slicer.util.messageBox("Success!")
-        self.updateApplyButtonState()
+            self._parameterNode.statusMsg = f"Registering bone '{self.currentBone.name}' in frame {target_frame}"
 
-    def onImportButton(self):
+            target_volume = AutoscoperMLogic.getItemInSequence(self._parameterNode.volumeSequence, target_frame)[0]
+            self.currentBone.setupFrame(target_frame, target_volume)
+
+            self.logic.registerBoneInFrame(
+                self.currentBone,
+                source_frame,
+                target_frame,
+                self._parameterNode.trackOnlyRoot
+            )
+
+            # If there is a next frame to register, propagate the transform generated from this frame
+            if target_frame != self._parameterNode.endFrame:
+                self.currentBone.copyTransformToNextFrame(target_frame)
+
+            # append all child bones to queue of bones to register next in this frame
+            if not self._parameterNode.trackOnlyRoot:
```

Comment:
Transform propagation to next child in TreeNode within the current frame doesn't seem to be working:

![image (1)](https://github.com/user-attachments/assets/65f6dcce-9a9a-43c1-8166-2e402f93c4a5)

Expected the mc1 to be in same relative alignment from tpm as in source scalar vol

Reply:
Will look into this. Is mc1 under rad or tpm in the hierarchy here?

Reply:
good question - I embedded a child or rad only - not a 3rd layer (i.e. not a child of tpm )

Reply:
Resolved in recent push.